typescript/db/src/repository/label.ts

```diff
@@ -1,77 +1,91 @@
+import { DbLabel, Repository } from "@labelflow/common-resolvers";
+import { LabelWhereInput } from "@labelflow/graphql-types";
+import { isEmpty } from "@labelflow/utils";
 import { Prisma } from "@prisma/client";
-import { Repository, DbLabel } from "@labelflow/common-resolvers";
+import { isNil } from "lodash/fp";
+import { Entry } from "type-fest";
 import { getPrismaClient } from "../prisma-client";
-import { castObjectNullsToUndefined } from "./utils";
 import { checkUserAccessToDataset } from "./access-control";
+import { AuthorizationError } from "./authorization-error";
+import { castObjectNullsToUndefined } from "./utils";
 
-export const countLabels: Repository["label"]["count"] = async (
-  whereWithUser
-) => {
-  if (!whereWithUser) {
-    return 0;
-  }
-  const { user, ...where } = whereWithUser;
-  if (user?.id == null) {
-    return 0;
-  }
+const getIdFilter = (
+  id: LabelWhereInput["id"]
+): Pick<Prisma.LabelWhereInput, "id"> | undefined => {
+  if (isNil(id)) return undefined;
+  // Make sure that no element is null or empty to satisfy Prisma constraints
+  const filtered = id.in.filter((item) => !isEmpty(item)) as NonNullable<
+    Entry<NonNullable<LabelWhereInput["id"]>["in"]>[1]
+  >[];
+  return isEmpty(filtered) ? undefined : { id: { in: filtered } };
+};
+
+type WhereWithUser =
+  | (LabelWhereInput & { user?: { id: string } })
+  | undefined
+  | null;
+
+const parseWhereInput = async (
+  whereWithUser: WhereWithUser
+): Promise<Prisma.LabelWhereInput | undefined> => {
+  if (isNil(whereWithUser)) return undefined;
+  const { user, id, ...where } = whereWithUser;
+  if (isEmpty(user?.id)) return undefined;
+  const idFilter = getIdFilter(id);
   if ("datasetId" in where) {
-    checkUserAccessToDataset({ where: { id: where.datasetId }, user });
-    return await (
-      await getPrismaClient()
-    ).label.count({
-      where: { image: { datasetId: where.datasetId ?? undefined } },
-    });
+    await checkUserAccessToDataset({ where: { id: where.datasetId }, user });
+    return {
+      ...idFilter,
+      image: { datasetId: where.datasetId ?? undefined },
+    };
   }
-  return await (
-    await getPrismaClient()
-  ).label.count({
-    where: castObjectNullsToUndefined({
-      ...where,
-      image: {
-        dataset: {
-          workspace: { memberships: { some: { userId: user?.id } } },
-        },
-      },
-    }),
+  return castObjectNullsToUndefined({
+    ...idFilter,
+    ...where,
+    image: {
+      dataset: { workspace: { memberships: { some: { userId: user?.id } } } },
```

Comment:
Use the `datasetid` here to simplify code, no need to do `checkUserAccessToDataset` anymore